reader.go

```diff
@@ -783,6 +787,59 @@ func (r *Reader) commitLoop(conn *Conn) func(stop <-chan struct{}) {
 	}
 }
 
+// partitionWatcher queries kafka and watches for partition changes, triggering a rebalance if changes are found.
+func (r *Reader) partitionWatcher(conn *Conn) func(stop <-chan struct{}) {
+	ticker := time.NewTicker(defaultPartitionWatchTime)
+	return func(stop <-chan struct{}) {
+		ops, err := conn.ReadPartitions(r.config.Topic)
+		if err != nil {
+			r.withErrorLogger(func(l *log.Logger) {
+				l.Printf("Problem getting partitions during startup, %v\n, Returning and setting up handshake", err)
+			})
+			return
+		}
+		// It's possible that the list of partitions returned are not in any order
+		// so put them into a map to compare.
+		oParts := make(map[int]struct{})
+		for _,p := range ops {
+			oParts[p.ID] = struct{}{}
+		}
+		for {
+			select {
+			case <-stop:
+				return
+			case <-ticker.C:
+				ops, err := conn.ReadPartitions(r.config.Topic)
+				if err != nil {
+					r.withErrorLogger(func(l *log.Logger) {
+						l.Printf("Problem getting partitions while checking for changes, %v\n", err)
+					})
+					return
+				}
+				nParts := make(map[int]struct{})
+				for _,p := range ops {
+					nParts[p.ID] = struct{}{}
+				}
+				if len(nParts) != len(oParts) {
+					r.withErrorLogger(func(l *log.Logger) {
+						l.Printf("Partition changes found, reblancing group: %v.", r.config.GroupID)
+					})
+					return
+				}
+
+				for i, _ := range nParts {
```

Comment:
This loop isn't necessary, you already looped through the `nParts` 11 lines above here.  Should be able to combine this loop and that loop together into a single loop.